caluma/extensions/events/form.py

```diff
@@ -0,0 +1,99 @@
+import csv
+import io
+import logging
+
+from django.db import transaction
+from django.db.models.signals import post_save
+from rest_framework.exceptions import ValidationError
+
+from caluma.caluma_core.events import on
+from caluma.caluma_form import models as caluma_form_models
+
+logger = logging.getLogger(__name__)
+
+
+@on(post_save, sender=caluma_form_models.AnswerDocument, raise_exception=True)
+@transaction.atomic
+def update_table_summary(instance, *args, **kwargs):
+    question = instance.answer.question
+    main_document = instance.answer.document
+
+    summary_question = question.meta.get("summary-question")
+    summary_mode = question.meta.get("summary-mode")
+    msg = f"Updating table summary: tq={instance.answer.question_id}"
+    logger.debug(msg)
+
+    if not summary_question or not summary_mode:
+        msg = f"Updating table summary: missing info in TQ meta: sq={summary_question},sm={summary_mode}"
+        raise ValidationError(msg)
+
+    msg = f"Updating table summary: sq={summary_question} sm={summary_mode}"
+    logger.debug(msg)
+
+    summary_answer, _ = caluma_form_models.Answer.objects.get_or_create(
+        document=main_document, question_id=summary_question
+    )
+
+    summary_modes = {"csv": _make_csv_summary}
+
+    summary_func = summary_modes.get(summary_mode)
+    if not summary_func:
+        return
+
+    summary_answer.value = summary_func(instance.answer)
+    summary_answer.save()
+
+
+@on(post_save, sender=caluma_form_models.Answer, raise_exception=True)
+@transaction.atomic
+def update_table_summary_from_row(instance, *args, **kwargs):
+    ad = caluma_form_models.AnswerDocument.objects.filter(
+        document=instance.document
+    ).first()
+    if not ad:
+        return
+
+    # AnswerDocument available, we're in a table
+    # Trigger by just saving the AnswerDocument
+    ad.save()
```

Comment:
I think we could avoid hitting the DB here by calling `update_table_summary()` directly

```suggestion
    # AnswerDocument available, we're in a table
    # Make sure the summaries (if any) are updated
   update_table_summary(instance=ad)
```

Reply:
Done.